fairlearn/postprocessing/_reject_option_classifier.py

```diff
@@ -0,0 +1,272 @@
+# Copyright (c) Fairlearn contributors.
+# Licensed under the MIT License.
+
+import numpy as np
+from sklearn import clone
+from sklearn.base import BaseEstimator, MetaEstimatorMixin
+from sklearn.exceptions import NotFittedError
+from sklearn.utils.validation import check_is_fitted
+from warnings import warn
+
+from ._threshold_operation import ThresholdOperation
+
+from ..utils._common import _get_soft_predictions
+from ..utils._input_validation import _validate_and_reformat_input
+from ._constants import (
+    BASE_ESTIMATOR_NONE_ERROR_MESSAGE,
+    BASE_ESTIMATOR_NOT_FITTED_WARNING)
+
+
+class RejectOptionClassifier(BaseEstimator, MetaEstimatorMixin):
+    r"""A classifier that produces group specific classifications inside a critical region.
+
+    :code:`RejectOptionClassifier` looks at the certainty by which binary classification
+    decisions are made. For decisions made with high certainty (= with a posterior
+    probability close to 0 or 1), the result of the classification remains the same.
+    For decisions with low certainty (= with a probability to get 1 close to 0.5), the
+    result is disregarded and instances from the favorable group receive the undesirable label,
+    whilst instances from the deprived group receive the desirable label. The exact range of
+    decisions considered to have low certainty is specified by the critical region.
+
```

Comment:
The original paper has the terminology that I don't think is best, and also `theta` parameter is confusing. So how about this:
* `deprived_group` -> `group_to_upselect`
* `favorable_group` -> `group_to_downselect`
* `desired_label` -> `selection_label`
Instead of `theta`, I would suggest something like:
* `critical_width` -- this would mean that the critical region is [0.5-cw/2, 0.5+cw/2]